Automattic/jetpack
# IP and WAF: Add CIDR support

## Proposed changes:

* Adds support to the `ip` package for handling CIDR ranges in all utility functions.
* Adds support to the `waf` package for handling CIDR ranges in the block and allow lists.

### Other information:


## Jetpack product discussion



https://github.com/Automattic/jetpack-scan-team/issues/1419

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






* Review and run unit tests added in this PR.
* Test the firewall and brute force protection features:
  * Test using both IPv4 and IPv6 addresses, with ranges using both hyphens and CIDR notation. See unit test for examples.
  * Test the firewall's block list.
  * Test that your IP address can be blocked by brute force protection.
  * Test the firewall's allow list for both the automatic firewall rules and brute force protection.

Easily testing CIDR notation: If your IP address is `127.0.0.1`, use `127.0.0.0/24` as a range that translates to `127.0.0.*`.
